src/Create/Create.tsx

```diff
@@ -3,12 +3,21 @@ import { useNavigate } from "react-router-dom";
 
 import { JournalEntry } from "../journal";
 import { useJournal } from "../JournalContext";
+import { useLocation } from 'react-router-dom';
 
 function JournalEntryForm(){
-
+    
     const { addJournalEntry } = useJournal();
 
-    const [journalEntry, setJournalEntry] = useState<JournalEntry>({
+    // for navigation back to home page after submitting
+    const navigate = useNavigate();
+
+    // for passing entry on edit
+    const location = useLocation();
+    const entry = location.state?.entry;
+    
+    const [journalEntry, setJournalEntry] = useState<JournalEntry>(
+        entry || {
```

Comment:
I did not notice this before and it has not been specified, the spacing is inconsistent in certain parts of the file. not something to worry about now but something that could be handled via code formater or linter of some kind.